dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java

```diff
@@ -714,4 +725,52 @@ private static void usage() {
         }
     }
 
+    /**
+     * Method to cross references handles of items that have been deleted with OAI core docs of said handle.
+     * If a match is found update that document.
+     * If no match is found create a document for persistence sake
+     *
+     * @param solrServer
+     *            The configured solr server to commit to
+     * @param handle
+     *            The handle to cross reference
+     *
+     */
+    private void updateDocsWithDeletes(SolrClient solrServer, String handle) {
+        try {
+            SolrQuery params = new SolrQuery("item.handle:" + handle);
+            SolrDocument result = SolrSearchQuerySingleSupportNull(solrServer, params);
+            if (result != null) {
+                solrServer.deleteByQuery(params.getQuery());
+            }
+            SolrInputDocument document = new SolrInputDocument();
+            document.setField("item.handle", handle);
+            document.setField("item.deleted", true);
```

Comment:
`item.public = true`  and `item.lastmodified = new Date()` are probably needed to expose this deleted document to GetRecord request